src/main/java/org/carlspring/cloud/storage/s3fs/S3AwsRegionProviderChain.java

```diff
@@ -0,0 +1,60 @@
+package org.carlspring.cloud.storage.s3fs;
+
+import java.util.function.Supplier;
+
+import software.amazon.awssdk.profiles.ProfileFile;
+import software.amazon.awssdk.regions.providers.AwsProfileRegionProvider;
+import software.amazon.awssdk.regions.providers.AwsRegionProviderChain;
+import software.amazon.awssdk.regions.providers.SystemSettingsRegionProvider;
+
+/**
+ * AWS Region provider that looks for the region in this order:
+ * <ol>
+ *   <li>Check the 'aws.region' system property for the region.</li>
+ *   <li>Check the 'AWS_REGION' environment variable for the region.</li>
+ *   <li>Check the {user.home}/.aws/credentials and {user.home}/.aws/config files for the region.</li>
+ * </ol>
+ */
```

Comment:
:1st_place_medal: for adding this javadoc! :)

Reply:
@carlspring / @ptirador By the way the `aws.region` property and `AWS_REGION` env var are going to be confusing since we already have `S3FS_REGION`. Aren't these going to conflict with each other?

Reply:
@steve-todorov When we obtain the region in `S3Factory`, these steps are made:
1. Check if the `s3fs.region` property exists. If so, return it.
2. If for any reason, it does not exist, then the custom region provider chain starts looking for the region from its providers.
3. If the custom region provider chain is not able to provide a region, then a default region value is returned (in this case, `us-east-1`).

![image](https://user-images.githubusercontent.com/6933533/106637122-9bdc6c00-6582-11eb-9d87-9b94e534493c.png)